server/src/main/java/io/crate/metadata/RelationName.java

```diff
@@ -49,11 +53,14 @@ public final class RelationName implements Writeable {
 
     public static RelationName of(QualifiedName name, String defaultSchema) {
         List<String> parts = name.getParts();
-        if (parts.size() > 2) {
+        if (parts.size() > 3) {
             throw new IllegalArgumentException(
-                "Table with more than 2 QualifiedName parts is not supported. Only <schema>.<tableName> works.");
+                "Table with more than 2 QualifiedName parts is not supported. Only <catalog>.<schema>.<tableName> works.");
+        }
+        if (parts.size() == 3) {
+            validateCatalogName(parts.get(0));
         }
-        if (parts.size() == 2) {
+        if (parts.size() >= 2) {
             return new RelationName(parts.get(0), parts.get(1));
         }
```

Comment:
Wouldn't this use the catalog as schema name?

Reply:
Thx! I changed a test in `SchemasTest` to catch that, should I maybe introduce a `RelationNameTest` ?